src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+const firstPromise = () => {
+  return new Promise((resolve, reject) => {
+    document.body.addEventListener('click', () => {
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+    }, 3000);
+  });
+};
+
+const secondPromise = () => {
+  return new Promise((resolve) => {
+    document.body.addEventListener('click', () => {
+      resolve('Second promise was resolved');
+    });
+
+    document.body.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+      resolve('Second promise was resolved');
+    });
```

Comment:
In the second promise, both click and contextmenu events resolve the promise. This could lead to multiple resolutions if both events are triggered. Consider removing the event listeners after the promise resolves to prevent this.